### DISCUSSION TITLE
Feature Request: Package-URL / PURL support

### DISCUSSION THREAD

**Discussion by mrueg** [curl/curl#12817] [Ideas]
I haven't found any request or any traces in the github discussions, issues and PRs for this, so please let me know if this got requested anywhere else.

```
A purl or package URL is an attempt to standardize existing approaches to reliably identify and locate software packages.

A purl is a URL string used to identify and locate a software package in a mostly universal and uniform way across programing languages, package managers, packaging conventions, tools, APIs and databases.

Such a package URL is useful to reliably reference the same software package using a simple and expressive syntax and conventions based on familiar URLs.
```
See: https://github.com/package-url/purl-spec

It would add a lot of convenience, if one could use curl to fetch purl-supported identifiers (or at least a subset of it)
e.g. 
```
curl --purl pkg:deb/debian/curl@7.50.3-1?arch=i386&distro=jessie
```
would allow curl to resolve this via purl-spec and fetch the according deb file.

**Comment by bagder** [comment]
I see nothing in this spec that specifies how there is a file transfer involved. A PURL is an identifier more than anything else. It does not say how the thing that is identified is transferred.

**Reply by mrueg** [reply]
Ah I guess you're right, I haven't checked the spec in that detail. I was simply assuming there's a mapping between the purl spec and a URL that can be generated from it. It looks like that is not the case. Sorry for the noise.
